src/ddmd/mars.d

```diff
@@ -1506,6 +1529,11 @@ Language changes listed by -transition=id:
     // So deps file generation should be moved after the inlinig stage.
     if (global.params.moduleDeps)
     {
+        StringTable importArray;
```

Comment:
Using a `StringTable` for this is rather clunky - it is designed for string interning which isn't actually necessary here - the `Module` instances themselves should already be unique.  The usual way to do this is to instead use `AA *` (`root.aav`) which uses the `Module` instance address as the hash.

Alternatively (and preferably) if this only needs to be run on `Modules` that haven't had semantic3 run, there is a member of `Module` that keeps track of which semantic has been run (`semanticRun`) which can be inspected to see if it has already been covered.  This will be much more efficient.